src/type-resolution/callables.ts

```diff
@@ -152,27 +155,48 @@ function inferInstanceMethodCall(
       `The type of ${logicalId} could not be inferred. Please provide the type explicitly.`
     );
   }
+}
 
-  function inferMethod(
-    typeRef: reflect.TypeReference,
-    call: ObjectLiteral
-  ): { method: reflect.Method; path: string } {
-    const methodPath = assertOneField(call.fields);
-    const [constructPath, methodName] = splitPath(methodPath);
-    const candidateType = resolveTypeFromPath(typeRef.type, constructPath);
-    const methods = candidateType.allMethods.filter((m) => !m.static);
-    const methodNames = methods.map((m) => m.name);
-
-    if (!methodNames.includes(methodName)) {
-      throw new Error(
-        `'${candidateType.fqn}' has no method called '${methodPath}'`
-      );
-    }
+function inferMethod(
+  typeRef: reflect.TypeReference,
+  call: ObjectLiteral
+): { method: reflect.Method; methodPath: string } {
+  const methodPath = assertOneField(call.fields);
+  const [constructPath, methodName] = splitPath(methodPath);
+  const candidateType = resolveTypeFromPath(typeRef.type, constructPath);
+  const methods = candidateType.allMethods.filter((m) => !m.static);
+  const methodNames = methods.map((m) => m.name);
+
+  if (!methodNames.includes(methodName)) {
+    throw new Error(
+      `'${candidateType.fqn}' has no method called '${methodPath}'`
+    );
+  }
 
-    return {
-      method: methods.find((m) => m.name === methodName)!,
-      path: methodPath,
+  return {
+    method: methods.find((m) => m.name === methodName)!,
+    methodPath,
+  };
+}
+
+function argsFromCall(
+  method: reflect.Method,
+  methodName: string,
+  fields: Record<string, TemplateExpression>
+): TypedArrayExpression {
+  const value = fields[methodName];
+  if (value.type === 'object') {
```

Comment:
Keyword arguments are still allowed. We can remove this in a subsequent PR if we decide to drop support for this.